CondFormats/HcalObjects/interface/HcalRecoParamWithPulseShapeT.h

```diff
@@ -17,7 +17,55 @@ class HcalRecoParamWithPulseShapeT {
     using PulseShapeConstView = typename PulseShapeCollection::ConstView;
     constexpr ConstView(RecoParamConstView recoView, PulseShapeConstView psView)
         : recoParamView_{recoView}, pulseShapeView_{psView} {};
-    constexpr float pulseShape(int id) const { return 0; }
+
+    constexpr float compute_pulse_shape_value(uint32_t const hashedId,
+                                              float const pulse_time,
+                                              int const sample,
+                                              int const shift) {
+      auto const recoPulseShapeId = this->recoParamView().ids()[hashedId];
+      auto const& acc25nsVec = this->pulseShapeView().acc25nsVec()[recoPulseShapeId];
+      auto const& diff25nsItvlVec = this->pulseShapeView().diff25nsItvlVec()[recoPulseShapeId];
+      auto const& accVarLenIdxMinusOneVec = this->pulseShapeView().accVarLenIdxMinusOneVec()[recoPulseShapeId];
+      auto const& diffVarItvlIdxMinusOneVec = this->pulseShapeView().diffVarItvlIdxMinusOneVec()[recoPulseShapeId];
+      auto const& accVarLenIdxZeroVec = this->pulseShapeView().accVarLenIdxZEROVec()[recoPulseShapeId];
+      auto const& diffVarItvlIdxZeroVec = this->pulseShapeView().diffVarItvlIdxZEROVec()[recoPulseShapeId];
```

Comment:
I'd suggest
```c++
auto const& pulseShape = this->recoParamView(recoPulseShapeId);
auto const& acc25nsVec = pulseShape.acc25nsVec();
...
```

Reply:
Done in 11aa37463e89e26d17c31a9e9d76e6f799b8a298